auth_oidc/controllers/main.py

```diff
@@ -48,3 +51,18 @@ def list_providers(self):
                     provider["auth_endpoint"], url_encode(params)
                 )
         return providers
+
+
+class OpenIDLogout(Session):
+
+    @http.route("/web/session/logout", type="http", auth="none")
+    def logout(self, redirect="/web/login"):
+        user = request.env["res.users"].sudo().browse(request.session.uid)
+        if user.oauth_provider_id.id:
+            p = request.env["auth.oauth.provider"].sudo().browse(user.oauth_provider_id.id)
+            if p.logout_endpoint:
+                r = request.env['ir.config_parameter'].sudo().get_param('web.base.url') + redirect
+                logout_url = p.logout_endpoint + "?client_id=" + p.client_id + "&post_logout_redirect_uri=" + r
```

Comment:
Yes `r` should be encoded when added as post_logout_redirect_uri. (e.g. when redirect is `/web/login?db=devel&something=different` it would possibly end up as parameter something on the logout_endpoint)

Reply:
Makes sense, I will update the PR